src/core/geometry/qgspoint.h

```diff
@@ -46,8 +46,58 @@ class CORE_EXPORT QgsPoint: public QgsAbstractGeometry
 
     /**
      * Construct a point with the provided initial coordinate values.
+     *
+     * If \a wkbType is set to `QgsWkbTypes::Point`, `QgsWkbTypes::PointZ`, `QgsWkbTypes::PointM` or `QgsWkbTypes::PointZM`
+     * the type will be set accordingly. If it is left to the default `QgsWkbTypes::Unknown`, the type will be set
+     * based on the following rules:
+     * - If only x and y are specified, the type will be a 2D point.
+     * - If any or both of the Z and M are specified, the appropriate type will be created.
+     *
+     * \code{.py}
+     *   pt = QgsPoint(43.4, 5.3)
+     *   pt.exportToWkt() # Point(43.4 5.3)
+     *
+     *   pt_z = QgsPoint(120, 343, 77)
+     *   pt.exportToWkt() # PointZ(120 343 77)
+     *
+     *   pt_m = QgsPoint(33, 88, m=5)
+     *   pt_m.m() # 5
+     *   pt_m.wkbType() # QgsWkbTypes.PointM
+     *
+     *   pt = QgsPoint(30, 40, wkbType=QgsWkbTypes.PointZ)
+     *   pt.z() # nan
+     *   pt.wkbType() # QgsWkbTypes.PointZ
+     * \endcode
      */
-    QgsPoint( double x = 0.0, double y = 0.0, double z = 0.0, double m = 0.0 );
+#ifndef SIP_RUN
+    QgsPoint( double x = 0.0, double y = 0.0, double z = std::numeric_limits<double>::quiet_NaN(), double m = std::numeric_limits<double>::quiet_NaN(), QgsWkbTypes::Type wkbType = QgsWkbTypes::Unknown ) SIP_SKIP;
+#else
+    QgsPoint( double x = 0.0, double y = 0.0, SIP_PYOBJECT z = Py_None, SIP_PYOBJECT m = Py_None, QgsWkbTypes::Type wkbType = QgsWkbTypes::Unknown ) [( double x = 0.0, double y = 0.0, double z = 0.0, double m = 0.0, QgsWkbTypes::Type wkbType = QgsWkbTypes::Unknown )];
+    % MethodCode
```

Comment:
cool, thanks for the feedback. Another question, why using 0.0 instead of NaN for the C++ signature?